arrows/ffmpeg/ffmpeg_video_input.cxx

```diff
@@ -49,65 +49,43 @@ class ffmpeg_video_input::priv
 {
 public:
   /// Constructor
-  priv() :
-    f_format_context(avformat_alloc_context()),
-    f_video_index(-1),
-    f_video_encoding(nullptr),
-    f_video_stream(nullptr),
-    f_frame(nullptr),
-    f_filtered_frame(nullptr),
-    f_packet(nullptr),
-    f_software_context(nullptr),
-    f_filter_graph(nullptr),
-    f_filter_sink_context(nullptr),
-    f_filter_src_context(nullptr),
-    f_start_time(-1),
-    f_backstep_size(-1),
-    f_frame_number_offset(0),
-    video_path(""),
-    filter_desc("yadif=deint=1"),
-    frame_advanced(false),
-    end_of_video(true),
-    number_of_frames(0),
-    collected_all_metadata(false),
-    estimated_num_frames(false),
-    sync_metadata(true)
+  priv()
   { }
 
   // f_* variables are FFmpeg specific
 
-  AVFormatContext* f_format_context;
-  int f_video_index;
-  AVCodecContext* f_video_encoding;
-  AVStream* f_video_stream;
-  AVFrame* f_frame;
-  AVFrame* f_filtered_frame;
-  AVPacket* f_packet;
-  SwsContext* f_software_context;
-  AVFilterGraph* f_filter_graph;
-  AVFilterContext *f_filter_sink_context;
-  AVFilterContext *f_filter_src_context;
+  AVFormatContext* f_format_context = avformat_alloc_context();
+  int f_video_index = -1;
+  AVCodecContext* f_video_encoding = nullptr;
+  AVStream* f_video_stream = nullptr;
+  AVFrame* f_frame = nullptr;
+  AVFrame* f_filtered_frame = nullptr;
+  AVPacket* f_packet = nullptr;
+  SwsContext* f_software_context = nullptr;
+  AVFilterGraph* f_filter_graph = nullptr;
+  AVFilterContext* f_filter_sink_context = nullptr;
+  AVFilterContext* f_filter_src_context = nullptr;
 
   // Start time of the stream, to offset the pts when computing the frame number.
   // (in stream time base)
-  int64_t f_start_time;
+  int64_t f_start_time = -1;
 
   // Presentation timestamp (in stream time base)
   int64_t f_pts;
 
   // Number of frames to back step when seek fails to land on frame before request
-  int64_t f_backstep_size;
+  int64_t f_backstep_size = -1;
 
   // Some codec/file format combinations need a frame number offset.
   // These codecs have a delay between reading packets and generating frames.
-  unsigned f_frame_number_offset;
+  unsigned f_frame_number_offset = 0;
 
   // Name of video we opened
-  std::string video_path;
+  std::string video_path = "";
```

Comment:
BTW, I don't believe this is necessary; a default-constructed string and a string initialized to an empty string literal should be equivalent in practice. (I'm not sure `std::string` even has `QString`s concept of a null string and an empty string being different, and even with `QString` you generally have to go out of your way to tell the difference, and doing so is frowned upon.)